src/main/java/com/aws/greengrass/logmanager/model/FileHashNamePair.java

```diff
@@ -0,0 +1,31 @@
+package com.aws.greengrass.logmanager.model;
+
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class FileHashNamePair {
+    // filehash -> filename
+    private Map<String, String> fileHashNamePairs = new ConcurrentHashMap<>();
+
+    public void updatePair(Path path) {
+        File folder = new File(path.toUri());
+        LogFile[] files = LogFile.of(folder.listFiles());
+        if (files.length != 0) {
+            for (LogFile file : files) {
+                String fileHash = file.hashString();
+                String fileName = file.getAbsolutePath();
+                fileHashNamePairs.putIfAbsent(fileHash, fileName);
+                if (fileHashNamePairs.get(fileHash) != fileName) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.